apix-impl/src/integration-test/java/eu/xenit/apix/tests/search/SearchServiceTest.java

```diff
@@ -151,6 +152,10 @@ public void TestSkip() throws IOException, InterruptedException {
         Assert.assertEquals(resultAll.getNoderefs().get(2), result.getNoderefs().get(1));
     }
 
+    @Ignore("Result of getTotalResultCount() is unreliable. It depends on the result of the getNumberFound() method of "
+            + "a search result set (see description of the method in interface ResultSetSPI). The result of the "
+            + "getNumberFound() method can have a different meaning depending on the search query consistency and the "
+            + "search engine being used.")
```

Comment:
This test and the other test at line 191 failed. I can make some changes to them so they succeed but I don't feel confident about them when I see this written in the description of the getNumberFound() interface method in ResultSetSPI:
/** * Attempt to get the number of rows that matched the query. This result set may only contain a section of the results. * If a skip count is given the number found may or may not include the items skipped. * This is best effort and only done if is cheap. * For SOLR it is cheap; for the DB it may be expensive as permissions are done post query. * If you want to know if there are more results to fetch use hasMore() * @return long */ public long getNumberFound();

getTotalResult (The method we're trying to test here) relies on that method. It depends on the query consistency and the search engine if skipped results are counted towards the total number of results or not. What should be done here if ignoring or removing the test is not ok.